File: TCS_Digital.java
import java.util.Scanner;
public class tcs {
	public static double pn(double x, int n) {
		if(n==0)
			return 1;
		if(n==1)
			return x;
		return ((2*n -1)*x*pn(x,n-1)-(n-1)*pn(x,n-2))/(n);
	}
	public static void main(String[] args) {
		Scanner s = new Scanner(System.in);
		int m = 2;
		m = s.nextInt();
		int n[] = new int[m];
		double x[] = new double[m];
		double res[] = new double[m];
		for(int i = 0;i<m && m<20;i++) {
			x[i] = s.nextDouble();
			n[i] = s.nextInt();	
		}
		for(int i = 0;i<m;i++) {
			res[i] = (Math.round(10000*(pn(x[i],n[i]))))/10000.0;
			//System.out.format("%.4f\n",res[i]);
			System.out.println(res[i]);			
		}		
	}
}

File: TCS_Digital.cpp
#include <iostream>
#include<cmath>
using namespace std;

double pn(double x, int n) {
		if(n==0)
			return 1;
		if(n==1)
			return x;
		return ((2*n -1)*x*pn(x,n-1)-(n-1)*pn(x,n-2))/(n);
}
int main()
{
    int m = 2;
    cin>>m;
    int n[m];
    double x[m], res[m];
    for(int i = 0;i<m && m<20;i++){
        cin>>x[i]>>n[i];
    }

    for(int i = 0;i<m && m<20;i++){
        res[i] = (round(pn(x[i],n[i])*10000))/10000.0;
        cout<<res[i]<<endl;
    }
    return 0;
}
